draft-ietf-jsonpath-base.md

```diff
@@ -1595,7 +1605,7 @@ the second argument is not a string conforming to {{-iregexp}}.
 | `$[?length(@.*) < 3]` | Invalid typing since `@.*` is a non-singular path |
 | `$[?count(@.*) == 1]` | Valid typing |
 | `$[?count(1) == 1]` | Invalid typing since `1` is not a path  |
-| `$[?count(foo(@.*)) == 1]` | Valid typing, where `foo` is a function extension with argument of type `OptionalNodesType` and result type `OptionalNodesType` |
+| `$[?count(foo(@.*)) == 1]` | Valid typing, where `foo` is a function extension with argument of type `NodelistType` and result type `NodelistType` |
 | `$[?match(@.timezone, 'Europe/.*')]`         | Valid typing |
 | `$[?match(@.timezone, 'Europe/.*') == true]` | Valid typing |
```

Comment:
Shouldn't this now fail to type check? I thought `BooleanType` was not comparable to JSON true/false.